agent-packages/packages/hubspot/src/index.ts

```diff
@@ -420,4 +425,249 @@ export class HubspotService implements BaseService<HubspotConfig> {
+          to: { id: associatedObjectId },
+          types: [
+            {
+              associationCategory: AssociationSpecAssociationCategoryEnum.HubspotDefined,
+              associationTypeId: associationTypeIds[associatedObjectType as HubspotEntityType]
+            }
+          ]
+        });
+      }
+
+      const response = await this.client.crm.objects.tasks.basicApi.create(taskInput);
+
+      return {
+        success: true,
+        data: { taskId: response.id }
+      };
+    } catch (error) {
+      console.error('Error creating HubSpot task:', error);
+      return {
+        success: false,
+        error: error instanceof Error ? error.message : 'Failed to create HubSpot task'
+      };
+    }
+  }
+
+  async updateTask(params: UpdateTaskParams): Promise<UpdateTaskResponse> {
+    try {
+      const properties: Record<string, string> = {};
+
+      if (params.title) {
+        properties.hs_task_subject = params.title;
+      }
+      if (params.status) {
+        properties.hs_task_status = params.status;
+      }
+      if (params.priority) {
+        properties.hs_task_priority = params.priority;
+      }
+      if (params.taskType) {
+        properties.hs_task_type = params.taskType;
+      }
+      if (params.body) {
+        properties.hs_task_body = params.body;
+      }
+      if (params.dueDate) {
+        properties.hs_timestamp = params.dueDate;
+      }
+      if (params.ownerId) {
+        properties.hubspot_owner_id = params.ownerId;
+      }
+
+      const updateInput: SimplePublicObjectInput = { properties };
+
+      const response = await this.client.crm.objects.tasks.basicApi.update(
+        params.taskId,
+        updateInput
+      );
+
+      return {
+        success: true,
+        data: { taskId: response.id }
+      };
+    } catch (error) {
+      console.error('Error updating HubSpot task:', error);
+      return {
+        success: false,
+        error: error instanceof Error ? error.message : 'Failed to update HubSpot task'
+      };
+    }
+  }
+
+  async searchTasks(params: TaskSearchParams): Promise<SearchTasksResponse> {
+    try {
+      const filterGroups: Array<{
+        filters: Array<{ propertyName: string; operator: FilterOperatorEnum; value: string }>;
+      }> = [];
+
+      // Add keyword search filters if keyword is provided
+      if (params.keyword) {
+        filterGroups.push(
+          ...['hs_task_subject', 'hs_task_body'].map((property) => ({
+            filters: [
+              {
+                propertyName: property,
+                operator: FilterOperatorEnum.ContainsToken,
+                value: params.keyword!
+              }
+            ]
+          }))
+        );
+      }
+
+      // Add owner filter
+      if (params.ownerId) {
+        filterGroups.push({
+          filters: [
+            {
+              propertyName: 'hubspot_owner_id',
+              operator: FilterOperatorEnum.ContainsToken,
+              value: params.ownerId
```

Comment:
Why would we compare an id using ContainsToken?
wouldn't 1 than match for 1,10,11,21 etc?
```suggestion
              operator: FilterOperatorEnum.EQ,
              value: params.ownerId
```

Reply:
Right.